### ISSUE TITLE
Enhancement Requests

### ISSUE THREAD

**Issue by RadicalR** [bucanero/pkgi-ps3#72]
1) Allow a way to do page up and page down. (Skipping multiple lines) (Recommendation: Left - page up, Right - page down.)
2) Confirm a download is select if the RAP is missing. 
3) When downloading a game, ask user if they would like to queue associated DLC.
4) Ability to download game updates (future proofing when Sony doesn't fix the update issue)
5) Prompt user if they would like to apply update if there is a database update when starting PKGi-PS3.

Thanks for considering these.

**Comment by bucanero** [comment]
Currently I don't have free time to work on ps3 homebrew, but regarding these features 

For 2., right now if you press `Square` you can see details and it will show if there's RAP info or not.

Regarding 4., my original idea was to ping the official servers, and get the update XML; then the xml could be parsed and the updates queued if needed.

So, for example, for `BLUS30721`,
```sh
wget --no-check-certificate  "https://a0.ww.np.dl.playstation.net/tpl/np/BCUS98114/BCUS98114-ver.xml"
```

you get:

```xml
<?xml version='1.0' encoding='UTF-8'?>
<titlepatch status="alive" titleid="BLUS30721">
	<tag name="BLUS30721_T3" popup="true" signoff="true">
		<package version="01.02" size="34731232" sha1sum="d2c8b0ee685c166acdcd534bfe7cb73f9b10338f" url="http://b0.ww.np.dl.playstation.net/tppkg/np/BLUS30721/BLUS30721_T3/9472a41e0f952e95/UP0102-BLUS30721_00-PATCH02000000000-A0102-V0100-PE.pkg" ps3_system_ver="03.6000">
			<paramsfo>
				<TITLE>ASURA'S WRATH</TITLE>
			</paramsfo>
		</package>
	</tag>
</titlepatch>
```

**Comment by retrosapien** [comment]
Can that url format be used to fetch any games updates!?

I was just looking for methods of updating installed psn games on the ps3 and was surprised i couldn't do it on pgki.
I then looked to irisman, as i knew iso's could be updated, and came up short there as well. 
Aldostools provided me with this fantastic url;

https://ps3.aldostools.org/updates.html

Maybe that could be incorporated into pgki?

**Comment by retrosapien** [comment]
i just tried with RaC full frontal assault.
curl -k "https://a0.ww.np.dl.playstation.net/tpl/np/NPUA80642/NPUA80642-ver.xml"

This is quite interesting. 
I recently found a pc app that fetches updates, PS3-GUDT, but this could easily be put into a batch script.
It is probably what that program is doing in the background, anyway.

thanks for the info!

**Comment by bucanero** [comment]
> Can that url format be used to fetch any games updates!?

yes, that URL format is generic, it can be used for any content. If there's an update, you'll get an XML. (as long as servers are online)

> 4) Ability to download game updates 

This feature is now available in v1.2.0. It's not automatic; you need to press `Square` to get the item's details, and then you can press `Square` again to check online for additional updates packages.
If there are updates, they will be loaded in memory, and will be listed under the "Updates" category. (switch categories with L2/R2)
